web/client/components/styleeditor/RulesEditor.jsx

```diff
@@ -0,0 +1,370 @@
+/*
+ * Copyright 2020, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import { Glyphicon, FormControl as FormControlRB, FormGroup, Button as ButtonRB, Alert } from 'react-bootstrap';
+import Fields from './Fields';
+import uuidv1 from 'uuid/v1';
+import Toolbar from '../misc/toolbar/Toolbar';
+import { FilterBuilderPopover } from './FilterBuilder';
+import { ScaleDenominatorPopover } from './ScaleDenominator';
+import Symbolizer, { SymbolizerMenu } from './Symbolizer';
+import ClassificationSymbolizer from './ClassificationSymbolizer';
+import localizedProps from '../misc/enhancers/localizedProps';
+import tooltip from '../misc/enhancers/tooltip';
+import Message from '../I18N/Message';
+import getBlocks from './config/blocks';
+
+const Button = tooltip(ButtonRB);
+const FormControl = localizedProps('placeholder')(FormControlRB);
+
+export function Rule({
+    title,
+    tools,
+    errorId,
+    children
+}) {
+    return (
+        <li
+            className="ms-style-rule">
+            <div className="ms-style-rule-head">
+                <div className="ms-style-rule-head-info">
+                    {title}
+                </div>
+                <div className="ms-style-rule-head-tools">
+                    {tools}
+                </div>
+            </div>
+            {errorId && <Alert bsStyle="danger">
+                <Message msgId={errorId}/>
+            </Alert>}
+            <ul className="ms-style-rule-body">
+                {children}
+            </ul>
+        </li>
+    );
+}
+
+Rule.propTypes = {
+    title: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
+    tools: PropTypes.node,
+    errorId: PropTypes.string
+};
+
+function EmptyRules() {
+    return (
+        <div style={{
+            position: 'relative',
+            height: '100%',
+            display: 'flex',
+            justifyContent: 'center',
+            alignItems: 'center',
+            textAlign: 'center'
+        }}>
+            <div>
+                <Glyphicon glyph="exclamation-mark" style={{ fontSize: 150 }}/>
```

Comment:
I think the exlamation mark is too warring. 
I suggest to add an arrow that indicates how to add rules.
![image](https://user-images.githubusercontent.com/1279510/86787043-f7825d80-c064-11ea-8010-69adb50ecf6a.png)